### DISCUSSION TITLE
How to modify the style and content of the project

### DISCUSSION THREAD

**Discussion by landonzeng** [elsa-workflows/elsa-core#2439] [Q&A]
How to modify the style and content of the project, I checked the source code and did not find a page style file that can be modified

**Comment by sfmskywalker** [comment]
Indeed, the dashboard app wasn't designed with theme-ability in mind. However, you can still just include a custom CSS and target specific and/or generic HTML elements to modify their appearance such as fonts, colors etc.

If you need to target specific HTML elements for which no ID or specific CSS class is provided, then please feel free to submit a PR that adds this.

**Reply by landonzeng** [reply]
In which file do I change the language? I want to change English to Chinese because one of my clients requires it to be displayed in the specified language! Thanks!

**Reply by sfmskywalker** [reply]
Unfortunately, I didn't do a good job when I added the localization files. Right now, you would have to fork the project, update various `localization.ts` files for each component with Chinese translations, and submit a PR (which I'm happy to accept if you do).

But what we really need is to complete [this issue](https://github.com/elsa-workflows/elsa-core/issues/1359). Once completed, you can provide translations using your own files without having to add them to the designer project itself.

**Reply by landonzeng** [reply]
I can try to translate the localization.ts file first and then submit the PR, but I don't find the localization.ts file in the project, where do I change this file again please?

**Reply by sfmskywalker** [reply]
Sure, sounds good. You can find the localization.ts files in each component folder (under designer): 

https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio/src/components

You need to look in each subfolder. For example: https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio/src/components/screens/workflow-blueprint-viewer/elsa-workflow-blueprint-properties-panel

**Reply by landonzeng** [reply]
I didn't find elsa-workflows-studio after forking this folder

**Reply by sfmskywalker** [reply]
I'm not sure what you mean. Weren't you looking for the various `localization.ts` files to add the translations?

**Reply by landonzeng** [reply]
Yes, I was trying to find the various localization.ts files to add translations, but I didn't find localization.ts after forking the project

**Reply by sfmskywalker** [reply]
Make sure to do this:

1. Fork the project (done)
2. Clone your fork
3. Look for all `localization.ts` files.

If you can't find the files on your local clone then you missed a step. You can see for yourself that the files really do exist in this repo in the master branch. For example: https://github.com/elsa-workflows/elsa-core/blob/master/src/designer/elsa-workflows-studio/src/components/screens/workflow-blueprint-viewer/elsa-workflow-blueprint-properties-panel/localizations.ts

**Reply by landonzeng** [reply]
Is it because the filter was added to the .gitignore file

`
src/designer/elsa-workflows-studio/dist
src/designer/elsa-workflows-studio/www
src/designer/elsa-workflows-studio/loader
`
, so I don't have this file after forking

**Reply by sfmskywalker** [reply]
It's unrelated to the .gitignore file. The file you're looking for exists in subfolders of `src/designer/elsa-workflows/studio/src/components/` - notice that it does not include `dist`. The dist folder is created when you build or run the Stencil project.

**Reply by landonzeng** [reply]
![微信图片_20211123101226](https://user-images.githubusercontent.com/7336781/142961841-32c9edee-0cd2-4737-b7fe-40bfca28579d.png)
I didn't find the designer/elsa-workflows-studio folder in the src directory after I forked the project

**Reply by sfmskywalker** [reply]
Take a look at the `master' branch:

![image](https://user-images.githubusercontent.com/938393/143230354-005141e1-81fc-4552-9421-a9ae47db2061.png)

**Reply by landonzeng** [reply]
![image](https://user-images.githubusercontent.com/7336781/143261135-7fe52bcd-5f31-4e92-a3a7-d3c95a58fe57.png)
The designer folder still doesn't exist. 
Am I doing something wrong?
I've added other languages to my local area, but if that's not possible, I would like to send these files to you by email

**Reply by MrGeneric** [reply]
@landonzeng Your fork looks way out of date:
![image](https://user-images.githubusercontent.com/3923223/143277061-b2c1ac2a-b64d-4450-be06-defd027d58a2.png)

You should see the designer folder here:
https://github.com/elsa-workflows/elsa-core/tree/master/src

**Reply by sfmskywalker** [reply]
@landonzeng You're cloning your fork, which seems to be almost 2 years behind compared to the Elsa Core main repository :)

When you update your fork then you'll see the designer folder.

**Reply by landonzeng** [reply]
Ok, thank you very much for your response, I will submit the additions to PR later @sfmskywalker  @MrGeneric

**Reply by landonzeng** [reply]
[https://github.com/elsa-workflows/elsa-core/pull/2523](https://github.com/elsa-workflows/elsa-core/pull/2523)
I have submitted multilingual files,Please review  @sfmskywalker

**Reply by landonzeng** [reply]
I have a question, how do I use zh-cn in my project?Is it changing to in index.html?

**Reply by sfmskywalker** [reply]
See https://github.com/elsa-workflows/elsa-core/pull/2523#issuecomment-979049898

**Reply by landonzeng** [reply]
I changed part of the translation and changed zh-cn to zh-CN 
[https://github.com/elsa-workflows/elsa-core/pull/2528](https://github.com/elsa-workflows/elsa-core/pull/2528)
